src/almanack/metrics/data.py

```diff
@@ -610,3 +629,53 @@ def _get_almanack_version() -> str:
         import almanack
 
         return almanack.__version__
+
+
+def fetch_ecosystems_repository_data(remote_url: Optional[str]) -> dict:
+    """
+    Fetch repository data from the repos.ecosyste.ms API
+    based on the remote URL.
+
+    Args:
+        remote_url (Optional[str]):
+            The remote URL of the repository to look up.
+
+    Returns:
+        dict:
+            The JSON response from the API as a dictionary.
+
+    Raises:
+        requests.RequestException:
+            If the API call fails or the response cannot
+            be parsed as JSON.
+    """
+
+    # check if we have no remote_url
+    if remote_url is None:
+        return {}
+
+    # Base API endpoint
+    api_endpoint = "https://repos.ecosyste.ms/api/v1/repositories/lookup"
+
+    # Encode the remote URL for the query parameter
+    encoded_url = quote(remote_url, safe="")
+
+    # Construct the full API URL
+    full_url = f"{api_endpoint}?url={encoded_url}"
+
+    try:
+        # Perform the GET request
+        response = requests.get(
+            full_url, headers={"accept": "application/json"}, timeout=120
```

Comment:
any reason to use 120?

Reply:
Thanks for double checking - we failed a check with Bandit security linting which suggested we use a timeout of some kind. I thought 1 minute felt too short and 2 felt okay. I'll bump to 5 minutes just in case there's some kind of backoff or query limitation from the API's we work with through this function.